packages/lab/src/slider/internal/SliderThumb.tsx

```diff
@@ -1,82 +1,194 @@
+      setIsTooltipVisible(false);
+    };
 
-  const handleFocus = () => {
-    setFocussed(true);
-    if (index !== null) setActiveThumb(index);
-  };
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (event.key === "Escape") {
+        setIsTooltipVisible(false);
+      }
+    };
 
-  const handleBlur = () => {
-    setFocussed(false);
-    setActiveThumb(undefined);
-  };
+    const handleKeydownOnThumb = (event: React.KeyboardEvent) => {
+      let newValue = value;
 
-  const showTooltip = focussed || activeThumb === index;
+      switch (event.key) {
+        case "ArrowRight":
+        case "ArrowUp":
+          newValue = Math.min(value + step, max);
+          break;
+        case "ArrowLeft":
+        case "ArrowDown":
+          newValue = Math.max(value - step, min);
+          break;
+        case "PageUp":
+          newValue = Math.min(value + step * stepMultiplier, max);
+          break;
+        case "PageDown":
+          newValue = Math.max(value - step * stepMultiplier, min);
+          break;
+        case "Home":
+          newValue = min;
+          break;
+        case "End":
+          newValue = max;
+          break;
+        default:
+          return;
+      }
 
-  return (
-    <div
-      style={{ left: `${thumbPosition}%` }}
-      className={withBaseName("container")}
-      onPointerOver={handlePointerOver}
-      onFocus={handleFocus}
-      onBlur={handleBlur}
-    >
+      event.preventDefault();
+
+      handleInputChange({
+        target: { value: newValue.toString() },
+      } as ChangeEvent<HTMLInputElement>);
+    };
+
+    return (
       <div
-        className={clsx(withBaseName("tooltip"), {
-          [withBaseName("tooltip-visible")]: showTooltip,
+        className={clsx(withBaseName(), {
+          [withBaseName("focused")]: isFocused,
+          [withBaseName("disabled")]: disabled,
+          [withBaseName("dragging")]: trackDragging,
+          [withBaseName("secondThumb")]: index === 1,
         })}
-        aria-expanded={showTooltip}
+        data-testid="sliderThumb"
+        onPointerEnter={handlePointerEnter}
+        onPointerLeave={handlePointerLeave}
+        onPointerDown={handlePointerDown}
+        style={{ left: offsetPercentage }}
       >
-        <svg
-          className={withBaseName("tooltip-arrow")}
+        <SliderTooltip
+          value={formattedValue}
+          isVisible={(isTooltipVisible || trackDragging) && !disabled}
+        />
+        <input
+          disabled={disabled}
+          type="range"
+          ref={inputRef}
+          onFocus={handleFocus}
+          onBlur={handleBlur}
+          className={withBaseName("input")}
+          value={value}
+          onChange={handleInputChange}
+          onKeyDown={handleKeydownOnThumb}
+          aria-labelledby={ariaLabelledBy}
+          aria-valuenow={value}
+          aria-valuetext={ariaValueText}
+          aria-label={ariaLabel}
+          aria-describedby={`saltSlider-${id}-${index}`}
```

Comment:
very minor, if we compute an `id`, do it once and pass down the same. or it can be broken later easily